test/name_lookup_test.py

```diff
@@ -502,6 +502,44 @@ def test_function_lookup_via_array_element(self):
         found = find_name("a[2].f", scope)
         self.assertIsNotNone(found)
 
+    def test_need_for_temporary_flattening(self):
+        """A case @jackvreeken came up with in review comments of #307
+
+        It's a case where ast.Class.find_class fails and tree.find_name
+        works, even without "temporary flattening" mentioned below.
+
+        Excerpt from outline of MLS:
+        3. If `A` is a Class:
+            1. `A` is temporarily flattened without modifiers
+        @jackvreeken wrote: Is there an example of this that used to fail?
+        E.g. extending from a class
+        that is part of contents of an extended class. Something like
+        ...
+        """
+        txt = """
+        // Added encapsulated to fix name lookup error detected by find_name
+        class A
+            // Doesn't have a class B itself, but gets one via C
+            extends C(B.bla=2);
```

Comment:
Our original discussion was about name lookup only, which is what the test case was meant to check, using the *new* name lookup with the *old* flattening code. Getting into instantiation in your comment, you appear to be looking at the *old* instantiation/flattening code by your reference to `flatten_symbols(instance_tree)`. 

The original test case (new name lookup, old flattening) I think correctly shows the answer as `bla = 1` because `1` is the topmost modification. If I use the new `tree.InstanceTree.instantiate()`, I also get `bla = 1`. Not sure why OpenModelica gives `bla = 2`, but we should look into it more to verify what is correct.

For your new test case in the comment, I only get the `value = 0` modification with the new instantiation, so it appears you have created another great test case that found another bug! 🎉